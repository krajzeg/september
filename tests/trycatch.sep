print("Executing the try.. block.")

try_return_value := try {
	missing = "Something"
	print("This part of try body should not be reached.")
} catch (EWrongType) {
	print("EWrongType should not be caught.")	
} catch (EMissingProperty) {
	print("EMissingProperty caught correctly.")
} finally {
	print("Finalizer executed correctly.")
}

print("Execution continued safely after catching exception.")
print("Try should return Nothing, and returned:")
print(try_return_value)
