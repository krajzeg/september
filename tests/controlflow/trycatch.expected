Executing the try.. block.
EMissingProperty caught correctly.
Finalizer executed correctly.
Execution continued safely after catching exception.
Try should return Nothing, and returned:
<Nothing>
